jelares/portfolio/src/main/java/com/google/sps/servlet-data/ExampleComments.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** Class containing numComments example comments */
+public final class ExampleComments {
```

Comment:
I like how you're thinking about encapsulation. Some thoughts:

Is `Example` meant to be in the class name? Does this represent comments that users may have written, or are these strictly "examples" that the website shows and are not editable?

If these represent all comments, including those added by users, consider removing "Example" from the name. Since this class represents a collection of x comments, consider naming it a "CommentBatch" or "CommentList". That said, it doesn't look like this class does more than an ArrayList, so you could consider just using an ArrayList instead, unless you have future plans for this class.

Reply:
I could use an ArrayList however I thought that having an ADT could be useful in case I wanted to add more features in the future.

Also, you make a good point about the name - the idea was that the class was made for the purpose of having example comments for the walkthrough, but in reality the class does generally represent a batch of comments - I will rename it to something more appropriate.

Reply:
To reiterate on this point, if time permits I plan on using this feature to store and retrieve blog-post entries on my portfolio from users. So in reality the ADT should represent any media object containing text field, author, title, date posted, etc. These are things which both comments and blog posts share, so I will rename the ExampleComments class to something along the lines of PostBatch

Reply:
An alternative would be to have a model object that represents a single comment or blog post, and let ArrayList handle collection-type things. For example:

```
class TextContent {
String text;
...
}

class Comment extends TextContent {
}

class BlogPost extends TextContent {
}
```

Storing and retrieving should probably be the responsibility of some other object(s) (see https://en.wikipedia.org/wiki/Single-responsibility_principle):

```
class CommentFetcher {
function ArrayList<Comment> fetchComments();
}
```

Or to generalize a bit you could have an entity fetcher with a parser object:

```
interface EntityParser {
function Comment parseEntity(Entity);
}

class CommentParser implements EntityParser {
}

class BlogPostParser implements EntityParser {
}

class EntityFetcher {
function Object fetch(String entityName, EntityParser parser);
}

```

Reply:
Will keep this in mind for our future projects, as we talked about on gchat